sys/net/gnrc/sock/ip/gnrc_sock_ip.c

```diff
@@ -120,8 +120,9 @@ ssize_t sock_ip_recv(sock_ip_t *sock, void *data, size_t max_len,
         return -EPROTO;
     }
     memcpy(data, pkt->data, pkt->size);
+    int pkt_size = (int)pkt->size;
```

Comment:
You could reuse `res` here.

Reply:
It would be also possible to let gnrc_sock_recv return the size of the paket (atm it return 0 at the end) and then just return res in the specific _recv function. What would you prefere?

Reply:
I think it is less intrusive (and easier to follow later on), if we do just for now 

```suggestion
    res = (int)pkt->size;
```

and do your proposed optimization in a follow-up. This would also allow us to bring this into the newest release, as this is an easy to follow fix. I'm not sure if the other solution would be.